src/win/tty.c

```diff
@@ -57,11 +57,17 @@
 
 static void uv_tty_capture_initial_style(CONSOLE_SCREEN_BUFFER_INFO* info);
 static void uv_tty_update_virtual_window(CONSOLE_SCREEN_BUFFER_INFO* info);
+static int uv__cancel_read_console(uv_tty_t* handle);
 
 
 /* Null uv_buf_t */
 static const uv_buf_t uv_null_buf_ = { 0, NULL };
 
+static CRITICAL_SECTION uv__read_console_trap_lock;
+static volatile BOOL uv__read_console_trap = FALSE;
+static volatile BOOL uv__read_console_started = FALSE;
+static HANDLE uv__read_console_done_event = INVALID_HANDLE_VALUE;
```

Comment:
On second thought, I am not sure I understand the suggestion about uv_tty_y, and I couldn’t find a similar pattern in libuv. Do you want me to put those global variables into a singleton struct instance?